ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -281,18 +283,23 @@ private KeyField findKeyField(final List<SelectExpression> selectExpressions) {
               = (DereferenceExpression) toExpression;
 
           if (SchemaUtil.matchFieldName(keyField, dereferenceExpression.toString())) {
-            return Optional.of(new Field(toName, i, keyField.schema()));
+            found = Optional.of(new Field(toName, i, keyField.schema()));
+            break;
           }
         } else if (toExpression instanceof QualifiedNameReference) {
           final QualifiedNameReference qualifiedNameReference
               = (QualifiedNameReference) toExpression;
 
           if (SchemaUtil.matchFieldName(keyField, qualifiedNameReference.getName().getSuffix())) {
-            return Optional.of(new Field(toName, i, keyField.schema()));
+            found = Optional.of(new Field(toName, i, keyField.schema()));
+            break;
           }
         }
       }
-      return Optional.empty();
+
+      return found
```

Comment:
when is it possible for either of these to be true?

Reply:
`GROUP BY ROWKEY` and `GROUP BY ROWTIME` if I remember rightly. Both of which work in 5.3.